shark_turbine/kernel/wave/codegen.py

```diff
@@ -43,6 +104,120 @@ def _emit_function_call_node(self, node: fx.Node):
 
         handler(self, node)
 
+    def lookup_node_values(self, node: fx.Node) -> List[Value]:
+        assert NDEBUG or isinstance(node, fx.Node)
+        values = self._node_values.get(node)
+        if values is None:
+            values = [self.root_sig.resolve_by_reference(("node", node))]
+            self._node_values[node] = values
+        return values
+
+    def bind_node_proxy(
+        self, node: fx.Node, proxy: IRProxyValue, *, attrs: Optional[NodeAttrs] = None
+    ):
+        """Binds a node's result to a Python/IR proxy object."""
+        assert NDEBUG or (isinstance(node, fx.Node) and isinstance(proxy, IRProxyValue))
+        assert (
+            node not in self._node_values
+        ), f"Cannot rebind node {node}: already bound"
+        if attrs is not None:
+            attrs.store(node)
+        self._node_values[node] = [proxy]
+
+    # Default offset_fn always return 0. Typically would be overriden by
+    # hardware constraints later on.
+    def offset_fn(self, i):
+        return 0
+
+
+def gen_sympy_index(emitter: WaveEmitter, expr: sympy.Expr, stage: int) -> OpResult:
+    stack: list[OpResult] = []
+
+    # Induction var is accessed outside of the loop. Sets value to value of stage.
+    induction_var = sympy.symbols("ARG0")
```

Comment:
This is not going to be the induction variable and in general there could be many induction variables. I think you can get that information from the class in wave.py.

Reply:
Sounds good thanks